src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -313,18 +313,32 @@ public PutItemResult putItem(PutItemRequest putItemRequest) {
      * TODO: write Javadoc.
      */
     public QueryResult query(QueryRequest queryRequest) {
-        // map table name
-        queryRequest = queryRequest.clone();
         TableMapping tableMapping = getTableMapping(queryRequest.getTableName());
-        queryRequest.withTableName(tableMapping.getPhysicalTable().getTableName());
+
+        // map table name
+        final QueryRequest clonedQueryRequest = queryRequest.clone();
+        clonedQueryRequest.withTableName(tableMapping.getPhysicalTable().getTableName());
 
         // map query request
-        tableMapping.getQueryAndScanMapper().apply(queryRequest);
+        //clonedQueryRequest.setExpressionAttributeNames(Optional.ofNullable(clonedQueryRequest.getFilterExpression())
+        //    .map(s -> new HashMap<>(clonedQueryRequest.getExpressionAttributeNames())).orElseGet(HashMap::new));
+        //clonedQueryRequest.setExpressionAttributeValues(Optional.ofNullable(clonedQueryRequest.getFilterExpression())
+        //    .map(s -> new HashMap<>(clonedQueryRequest.getExpressionAttributeValues())).orElseGet(HashMap::new));
```

Comment:
Did you mean to leave this comments in here?

Reply:
for a bit, I was trying to make scan and query code similarish, so I could extract common method(s)..